packages/firestore/src/remote/watch_change.ts

```diff
@@ -449,17 +454,18 @@ export class WatchChangeAggregator {
       hashCount = 0
     } = unchangedNames;
 
-    // TODO(Mila): Remove this validation, add try catch to normalizeByteString.
-    if (typeof bitmap === 'string') {
-      const isValidBitmap = this.isValidBase64String(bitmap);
-      if (!isValidBitmap) {
-        logWarn('Invalid base64 string. Applying bloom filter failed.');
-        return false;
+    let normalizedBitmap: Uint8Array;
+    try {
+      normalizedBitmap = normalizeByteString(bitmap).toUint8Array();
+    } catch (err) {
+      if (err instanceof Base64DecodeError) {
+        logWarn('Base64 string error: ', err);
+      } else {
+        logWarn('Normalizing bloom filter bitmap failed: ', err);
```

Comment:
Just re-throw any other error here. We only want to "gracefully" handle base64 decoding errors; other errors should crash, since they should "never happen".